addons/project/models/project.py

```diff
@@ -945,18 +939,32 @@ def _compute_subtask_count(self):
 
     @api.onchange('company_id')
     def _onchange_task_company(self):
-        if self.project_id.company_id != self.company_id:
-            self.project_id = False
+        if self.effective_project_id.company_id != self.company_id:
+            self.effective_project_id = False
 
-    @api.depends('project_id.company_id')
+    @api.onchange('recurring_task')
+    def _onchange_recurring_task(self):
```

Comment:
Onchange methods are only trigger on the UI.

To convert into a compute method to apply the behavior on the whole business logic.

Reply:
Could also be a api.constraint to make the validation at the end.

Reply:
Indeed, it should be a constraint.